discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -77,16 +77,32 @@ def set_ingress_and_api_vips(self, controller):
     def start_install(self):
         self.api_client.install_cluster(cluster_id=self.id)
 
-    def wait_for_installing_in_progress(self):
+    def wait_for_installing_in_progress(self, nodes_count=1):
         utils.wait_till_at_least_one_host_is_in_status(
             client=self.api_client,
             cluster_id=self.id,
-            statuses=[consts.NodesStatus.INSTALLING_IN_PROGRESS]
+            statuses=[consts.NodesStatus.INSTALLING_IN_PROGRESS],
+            nodes_count=nodes_count
+        )
+
+    def wait_for_cluster_in_error_status(self):
+        utils.wait_till_cluster_is_in_status(
+            client=self.api_client,
+            cluster_id=self.id,
+            statuses=[consts.ClusterStatus.ERROR]
         )
 
     def cancel_install(self):
         self.api_client.cancel_cluster_install(cluster_id=self.id)
 
+    def get_bootstrap_node(self):
+        hosts = self.api_client.get_cluster_hosts(self.id)
+        for host in hosts:
+            if host["role"] == consts.NodeRoles.MASTER:
+                if host.get('bootstrap', True):
```

Comment:
host.get('bootstrap', True) defaults to True if 'bootstrap' key is not found. I'm not sure this is your intention